src/www/ui_tests/api/Controllers/MaintenanceControllerTest.php

```diff
@@ -190,4 +217,169 @@ public function testCreateMaintenance()
+
+    $mess = _("The maintenance job has been queued");
+
+    $this->maintagentPlugin->shouldReceive('handle')->withArgs([$body])->andReturn($mess);
+    $reqBody = $this->streamFactory->createStream(json_encode(
+      $this->rq
+    ));
+
+    $requestHeaders = new Headers();
+    $requestHeaders->setHeader('Content-Type', 'application/json');
+    $request = new Request("POST", new Uri("HTTP", "localhost"),
+      $requestHeaders, [], [], $reqBody);
+
+    $this->expectException(HttpForbiddenException::class);
+    $this->maintenanceController->createMaintenance($request, new ResponseHelper(), null);
+  }
+
+  /**
+   * @test
+   * -# Test MaintenanceController::testCreateMaintenance() for valid create maintenance
+   * -# Check if access is denied with HttpForbiddenException  for non admin users
+   */
+  public function testCreateMaintenanceWithBadRequest()
+  {
+    $_SESSION['UserLevel'] = 10;
+
+    $this->rq["options"] = [];
+
+    $reqBody = $this->streamFactory->createStream(json_encode(
+      $this->rq
+    ));
+    $requestHeaders = new Headers();
+    $requestHeaders->setHeader('Content-Type', 'application/json');
+    $request = new Request("POST", new Uri("HTTP", "localhost"),
+      $requestHeaders, [], [], $reqBody);
+
+    $this->expectException(HttpBadRequestException::class);
+    $this->maintenanceController->createMaintenance($request, new ResponseHelper(), null);
+  }
+
+
+  /**
+   * @test
+   * -# Test MaintenanceController::testCreateMaintenance() for valid create maintenance
+   * -# Check if  HttpNotFoundException is thrown when unknown key is encountered
+   */
+  public function testCreateMaintenanceOptionKeyNotFound()
+  {
+    $_SESSION['UserLevel'] = 10;
+
+    array_push($this->rq["options"],"M");
+    $alteredOptions = array();
+    foreach ($this->rq['options'] as $key) {
+      $alteredOptions[$key] = $key;
+    }
+    $body = $this->rq;
+    $body['options']  = $alteredOptions;
+
+    $this->maintagentPlugin->shouldReceive('getOptions')->andReturn($this->OPTIONS);
+
+    $mess = _("The maintenance job has been queued");
+
+    $this->maintagentPlugin->shouldReceive('handle')->withArgs([$body])->andReturn($mess);
+    $reqBody = $this->streamFactory->createStream(json_encode(
+      $this->rq
+    ));
+
+    $requestHeaders = new Headers();
+    $requestHeaders->setHeader('Content-Type', 'application/json');
+    $request = new Request("POST", new Uri("HTTP", "localhost"),
+      $requestHeaders, [], [], $reqBody);
+
+    $this->expectException(HttpNotFoundException::class);
+    $this->maintenanceController->createMaintenance($request, new ResponseHelper(), null);
+  }
+
+  /**
+   * @test
+   * -# Test MaintenanceController::testCreateMaintenance() for valid create maintenance
+   * -# Check if  HttBadRequestException is thrown when GoldDate is not provided
+   */
+  public function testCreateMaintenanceInvalidGoldDate()
+  {
+    $_SESSION['UserLevel'] = 10;
+
+    $this->rq["goldDate"] = "";
```

Comment:
Please create a local copy for the variable if modifying. Otherwise all other test cases will be effected if the test cases are run in different mode.

Reply:
Thank you @GMishx ! I have committed suggested changes.